src/coreclr/vm/stdinterfaces_wrapper.cpp

```diff
@@ -322,27 +74,13 @@ HRESULT __stdcall Unknown_QueryInterface(IUnknown* pUnk, REFIID riid, void** ppv
     }
     CONTRACTL_END;
 
+    if (!CanCallRuntimeInterfaceImplementations())
```

Comment:
This is definitely an improvement. Thoughts on using the HRESULT we have for runtime shutdown? I thought there was a `COR` error for that.

Reply:
I wanted to match the existing behavior to ensure we didn't somehow end up with a bad back-compat problem. I think the only `COR` error we have is `COR_E_APPDOMAINUNLOADED`, which the exception handling mechanism has some special code for (it's treated similarly to a stack overflow).

Reply:
Okay. If we previously would have returned `E_FAIL`, let's continue with that.